web/views.py

```diff
@@ -272,17 +147,178 @@ def post(self, request, *args, **kwargs):
+            raise Http404(_('No %(verbose_name)s found matching the query') %
+                          {'verbose_name': queryset.model._meta.verbose_name})
+        return obj
+
+    def post(self, request, *args, **kwargs):
+        '''
+        If deleteChild form submitted, mark child as deleted in the db.
+        '''
+        if 'deleteChild' in self.request.POST and self.request.method == 'POST':
+            child = self.get_object()
+            child.deleted = True
+            child.save()
+            return HttpResponseRedirect(self.get_success_url())
+        return super().post(request, *args, **kwargs)
+
+
+class ParticipantEmailPreferencesView(LoginRequiredMixin, generic.UpdateView):
+    """
+    Allows a participant to update their email preferences - when they can be contacted.
+    """
+    template_name = 'web/participant-email-preferences.html'
+    model = User
+    form_class = forms.EmailPreferencesForm
+
+    def get_object(self, queryset=None):
+        return self.request.user
+
+    def get_success_url(self):
+        return reverse('web:email-preferences')
+
+
+class StudiesListView(generic.ListView):
+    '''
+    List all active, public studies.
+    '''
+    template_name = 'web/studies-list.html'
+    model = Study
+
+    def get_queryset(self):
+        # TODO if we need to filter by study demographics vs user demographics
+        # TODO or by if they've taken the study before this is the spot
+        return super().get_queryset().filter(state='active', public=True)
+
+
+class StudyDetailView(generic.DetailView):
+    '''
+    Show the details of a study.  If the user has selected a child, they can
+    participate in the study and be forwarded/proxied to the js application
+    '''
+    template_name = 'web/study-detail.html'
+    model = Study
+
+    def get_queryset(self):
+        return super().get_queryset().filter(state='active', public=True)
+
+    def get_object(self, queryset=None):
+        '''
+        Needed because view expecting pk or slug, but url has UUID. Looks up
+        study by uuid.
+        '''
+        # Use a custom queryset if provided; this is required for subclasses
+        # like DateDetailView
+        if queryset is None:
+            queryset = self.get_queryset()
+
+        uuid = self.kwargs.get('uuid')
+
+        if uuid is not None:
+            queryset = queryset.filter(uuid=uuid)
+        try:
+            # Get the single item from the filtered queryset
+            obj = queryset.get()
+        except queryset.model.DoesNotExist:
+            raise Http404(_('No %(verbose_name)s found matching the query') %
+                          {'verbose_name': queryset.model._meta.verbose_name})
+        return obj
+
+    def get_context_data(self, **kwargs):
+        '''
+        If authenticated, add demographic presence, and children to context data dict
+        '''
+        context = super().get_context_data(**kwargs)
+        context['user'] = self.request.user
```

Comment:
We're using the `django.template.context_processors.request` context processor so the request should already be accessible in any template as `request`.